tools/sggolangcilint/tools.go

```diff
@@ -10,41 +10,63 @@ import (
 	"os/exec"
 	"path/filepath"
 	"runtime"
+	"text/template"
 
 	"go.einride.tech/sage/sg"
 	"go.einride.tech/sage/sgtool"
 )
 
 const (
 	name    = "golangci-lint"
-	version = "2.0.2"
+	version = "2.1.1"
+
+	RunRelativePathModeGitRoot    = "gitroot"
+	RunRelativePathModeGomod      = "gomod"
+	RunRelativePathModeCfg        = "cfg" // WARNING: not recommended; paths will be relative to .sage/tools/golangci-lint
+	RunRelativePathModeWorkingDir = "wd"  // WARNING: not recommended according to official docs
 )
 
-//go:embed golangci.yml
-var DefaultConfig []byte
+type RunRelativePathMode string
+
+//go:embed golangci.yml.tmpl
+var configTemplate []byte
+
+// Config holds values which will be applied to the template, ultimately resulting in the final golangci.yml file.
+type Config struct {
+	// The mode used to evaluate relative paths. It's used by exclusions, Go plugins and some linters.
+	RunRelativePathMode RunRelativePathMode
+	// Which file paths to exclude from issue reporting. The paths will still be analyzed.
+	// Golangci-Lint will:
+	// - replace "/" with the current OS file path separator to properly work on Windows.
+	// - allow you to optionally use regexps here, like ".*\\.my\\.go$".
+	// - treat paths relative to the setting of RunRelativePathMode.
+	LintersExclusionsPaths []string
+}
 
-func Command(ctx context.Context, args ...string) *exec.Cmd {
-	sg.Deps(ctx, PrepareCommand)
+func Command(ctx context.Context, config Config, args ...string) *exec.Cmd {
+	sg.Deps(ctx, func(ctx context.Context) error {
+		return PrepareCommand(ctx, config)
+	})
 	return sg.Command(ctx, sg.FromBinDir(name), args...)
 }
 
 func defaultConfigPath() string {
 	return sg.FromToolsDir(name, ".golangci.yml")
 }
 
-func CommandInDirectory(ctx context.Context, directory string, args ...string) *exec.Cmd {
+func CommandInDirectory(ctx context.Context, config Config, directory string, args ...string) *exec.Cmd {
 	configPath := filepath.Join(directory, ".golangci.yml")
 	if _, err := os.Lstat(configPath); errors.Is(err, os.ErrNotExist) {
 		configPath = defaultConfigPath()
 	}
 	cmdArgs := append([]string{"run", "--allow-parallel-runners", "-c", configPath}, args...)
-	cmd := Command(ctx, cmdArgs...)
+	cmd := Command(ctx, config, cmdArgs...)
 	cmd.Dir = directory
 	return cmd
 }
 
 // Run GolangCI-Lint in every Go module from the root of the current git repo.
-func Run(ctx context.Context, args ...string) error {
```

Comment:
This will be a breaking change for all clients, would be nice if clients that are using the default config today, doesnt need to do anything, and for clients that are using custom config, can supply a config.
Could maybe add `RunWithConfig`?

Reply:
I guess that would mean adding a new function for each existing function, right?
That would be `RunWithConfig`, `FixWithConfig`, `CommandWithConfig`, `CommandInDirectoryWithConfig`, `PrepareCommandWithConfig`.

And then I guess you would want the old `Run`, `Fix` etc hard-code a config that would be used, or did you want to keep using the old config file?

Technically, the bump to v2 _is_ a breaking change also for the CLI arguments which clients may be passing. We could add a whole new `sggolangcilintv2` package instead, and you can keep using this `sggolangcilint` (v1) as-is for as long as you want?

Reply:
I've now created a `sggolangcilintv2` package instead. Let me know what you think.

Reply:
I think that is better, then clients can choose when to update :+1: So marking it as deprecated is good, so we eventually can remove it.

Reply:
Ok, now I have also deprecated `sggolangcilint`. Let me know if you are missing anything.

Reply:
Tested it in pod-gateway, then it fails with deprecation warning :man_facepalming: maybe best to skip that initially then?
```
[sggolangcilint:prepare-command] fetching https://github.com/golangci/golangci-lint/releases/download/v1.64.8/golangci-lint-1.64.8-linux-amd64.tar.gz ...
[go-lint] main.go:132:9: SA1019: sggolangcilint.Run is deprecated: Use `sggolangcilintv2.Run` instead. Run GolangCI-Lint in every Go module from the root of the current git repo. (staticcheck)
[go-lint]       return sggolangcilint.Run(
```

Reply:
Yeah, that was expected 😄 but sure, I removed that last commit now.